profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -7,36 +7,51 @@
   Some references used:
   https://claritydev.net/blog/the-most-common-mistakes-when-using-react
 */
-import { useEffect } from "react";
+import { useCallback, useContext, useEffect } from "react";
 
+// This was missing a return but that showButton could be abstracted out.
 export function FunctionsAsComponents({ buttonText = "Start Now" }) {
   const showButton = () => {
-    <button>{buttonText}</button>;
+    return <button>{buttonText}</button>;
   };
 
   return <div>{showButton()}</div>;
 }
 
 export function objectShallowCopying() {
   const object = { a: { c: 1 }, b: 2 };
-  const copy = { ...object };
+  const copy = structuredClone(object);
   copy.a.c = 2;
   return { ...object };
 }
 
 export function arrayShallowCopying() {
   const array = [{ a: 1 }, { a: 2 }, { a: 3 }];
-  const copy = [...array];
+  const copy = structuredClone(array);
   return copy;
 }
 
+// fetchURL changes forces a new fetch. And the main problem with this one
+// is mostly likely to be caused by parent behavior/changes.
+// Maybe using a ref to compare values instead
 export function UseEffectThrashing({ fetchURL, label }) {
+  const prevValue = useRef(fetchURL);
   useEffect(() => {
-    const fetchData = async () => {
-      await fetch(fetchURL);
-    };
+    // If previous value is same as fetchURL, then do nothing.
+    if (prevValue.current === fetchURL) return;
 
-    fetchData();
+    const debounce = setTimeout(() => {
```

Comment:
The key is using an abortController, not debouncing. Debouncing changes the behavior of this component as is, while abort controller does not.